src/app/MarineApp/components/MetricsContext/MetricsContext.tsx

```diff
@@ -0,0 +1,48 @@
+import { createContext, FC, useCallback, useContext, useEffect, useMemo, useState } from "react"
+
+type notifyParams = {
+  widgetName: string
+  visible: boolean
+}
+
+type MetricsContextValue = {
+  elementsVisible?: boolean
+  notifyVisibility?: (arg: notifyParams) => void
+}
+
+export const MetricsContext = createContext<MetricsContextValue>({})
+
+export const MetricsProvider: FC = ({ children }) => {
+  const [visibleElements, setVisibleElements] = useState<Array<string>>([])
+
+  const notifyVisibility = useCallback((element: notifyParams) => {
+    if (element.visible) {
+      setVisibleElements((prevArr) => {
+        if (prevArr.includes(element.widgetName)) {
```

Comment:
I don't think `prevArr` is a very descriptive name. `currentVisibleElements` or something like that?